grace/server.go

```diff
@@ -187,11 +191,15 @@ func (srv *Server) handleSignals() {
 		default:
 			log.Printf("Received %v: nothing i care about...\n", sig)
 		}
-		srv.signalHooks(PostSignal, sig)
+		go srv.signalHooks(PostSignal, sig)
+		srv.wg.Done()
 	}
 }
 
 func (srv *Server) signalHooks(ppFlag int, sig os.Signal) {
+	srv.wg.Add(1)
```

Comment:
If we don't add the wait group, then we have a change which the process exit before any pre/post handler get called.